google/resource_google_folder.go

```diff
@@ -0,0 +1,160 @@
+package google
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	resourceManagerV2Beta1 "google.golang.org/api/cloudresourcemanager/v2beta1"
+	"strings"
+)
+
+func resourceGoogleFolder() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGoogleFolderCreate,
+		Read:   resourceGoogleFolderRead,
+		Update: resourceGoogleFolderUpdate,
+		Delete: resourceGoogleFolderDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceGoogleFolderImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			// Format is 'folders/{folder_id}.
+			// The folder id holds the same value.
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			// Format is either folders/{folder_id} or organizations/{org_id}.
+			"parent": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			// Must be unique amongst its siblings.
+			"display_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"lifecycle_state": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"create_time": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceGoogleFolderCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	displayName := d.Get("display_name").(string)
+	parent := d.Get("parent").(string)
+
+	op, err := config.clientResourceManagerV2Beta1.Folders.Create(&resourceManagerV2Beta1.Folder{
+		DisplayName: displayName,
+	}).Parent(parent).Do()
+
+	if err != nil {
+		return fmt.Errorf("Error creating folder '%s' in '%s': %s", displayName, parent, err)
+	}
+
+	err = resourceManagerV2Beta1OperationWait(config, op, "creating folder")
+
+	if err != nil {
+		return fmt.Errorf("Error creating folder '%s' in '%s': %s", displayName, parent, err)
+	}
+
+	// Since we waited above, the operation is guaranteed to have been successful by this point.
+	waitOp, err := config.clientResourceManager.Operations.Get(op.Name).Do()
+	if err != nil {
+		return fmt.Errorf("The folder '%s' has been created but we could not retrieve its id. Delete the folder manually and retry or use 'terraform import'.", displayName)
```

Comment:
The err is eaten here; should probably include it in the error message

Reply:
Fixed.